neurodamus/io/sonata_config.py

```diff
@@ -219,17 +158,19 @@ def _extract_circuits_info(self) -> dict:
         """
         node_info_to_circuit = {"nodes_file": "CellLibraryFile", "type": "PopulationType"}
 
-        if "node_set" not in self._entries:
+        simulation_nodeset_name = self._sim_conf.node_set or ""
+        if not simulation_nodeset_name:
             logging.warning("Simulating all populations from all node files...")
-        network = self._circuit_networks
+
+        network = json.loads(self._circuit_conf.expanded_json)["networks"]
 
         def make_circuit(nodes_file, node_pop_name, population_info):
             if not os.path.isabs(nodes_file):
-                nodes_file = os.path.join(os.path.dirname(self.network), nodes_file)
+                nodes_file = os.path.join(os.path.dirname(self._sim_conf.network), nodes_file)
```

Comment:
We could take this change to change to `pathlib.Path`.

Reply:
I actually have a separate PR that fixes a bunch of this; libsonata returns absolute paths, so it's not necessary.

Reply:
I thought we would use pathlib.Path to replace `os.path` everywhere, non? We need to agree on the convention. If you will replace them in another PR, that's also fine.

Reply:
yeah, we should agree and do it in a different PR, IMO.

Reply:
ok, great!